float64 t
float64 q0
float64 q1
string path
bool finish